### DISCUSSION TITLE
Agile Nordpool Spot Rates not being picked up?

### DISCUSSION THREAD

**Discussion by stjohncanning** [springfall2008/batpred#970] [Q&A] [answered]
Can't seem to get the Agile Nordpool lookahead working.   Have added in the relevant code into the app.yaml:

futurerate_url: '<https://www.nordpoolgroup.com/api/marketdata/page/325?currency=GBP>'
futurerate_adjust_import: True
futurerate_adjust_export: False

The log shows accessing the relevant url:

15:19:05 INFO pred_bat: Fetching futurerate data from <https://www.nordpoolgroup.com/api/marketdata/page/325?currency=GBP>
15:19:05 INFO pred_bat: Download <https://www.nordpoolgroup.com/api/marketdata/page/325?currency=GBP>

But nothing changes on the graphs or the charge plan, and no other relevant comments in the log files that I can see..
I have another automation running which uses the same look-ahead url and that picks up the relevant data and shows the deltas.
What a I missing?

**Comment by springfall2008** [comment]
If it's working you will see a difference between 10am and 4pm when some of the rates for the next day will be estimated using Nordpool and will have a set of scales against them on the HTML plan as below:

? ⚖ - Rate that has been estimated using future rate estimation data (e.g. Nordpool)

**Comment by stjohncanning** [comment]
Thanks for reply.
Could not see the scales next to the rates, only the ‘?’, and the rates did
not change until after 16:00, whilst the other automation picked up the new
rates.      I’ll play around a bit more tomorrow,  if you have inputs on
anything I should be specifically looking at would be appreciated.

**Comment by springfall2008** [comment]
I just tried it and its working for me, can you re-test in case the site was down as the log should show the Nordpool rates just after the download message?

**Comment by stjohncanning** [comment]
Hi,
Was not able to look at this today before 16:00 as busy elsewhere.
However, logs from just now at 17:50 with futurerate active in the yaml,
and at 17:52 with it inactive.
Not sure what it should look like when it is working correctly if not
before 16:00?
I have been playing around with quite a few settings recently so maybe have
upset something (I have reloaded the defaults but maybe that was not
enough).  I had thought to install PREDBAT again after deleting all
relevant files to start off with a clean slate and a fresh apps.yaml as I
seem to be the only one with an issue!

Regards
StJohn

Futurerate Active

17:50:00 INFO pred_bat: Predbat mode is set to Control charge
17:50:00 INFO pred_bat: Found 21600 load_today datapoints going back 14 days
17:50:02 INFO pred_bat: Car charging hold True threshold 6.0
17:50:02 INFO pred_bat: Current data so far today: load 8.19 kWh import
11.8 kWh export 12.7 kWh pv 16.7 kWh
17:50:02 INFO pred_bat: Fetching futurerate data from <
https://www.nordpoolgroup.com/api/marketdata/page/325?currency=GBP>
17:50:02 INFO pred_bat: Standing charge is set to 60.01800000000001 p
17:50:03 INFO pred_bat: Rate min forward looking: now 10.58 at end of
forecast 10.58
17:50:03 INFO pred_bat: Import rates min 10.58 max 32.85 average 16.92
17:50:03 INFO pred_bat: Export rates min 15.0 max 15.0 average 15.0
17:50:03 INFO pred_bat: Rate thresholds (for charge/discharge) are import
32.35p (0.0) export 15.0p (0.0)
17:50:03 INFO pred_bat: High export rate found rates in range 15.0 to 15.0
17:50:03 INFO pred_bat: Low Import rate found rates in range 11.02 to 32.01
17:50:03 INFO pred_bat: Todays energy import 11.8 kWh export 12.7 kWh cost
-49.37 p import 141.13 p export -190.5 p
17:50:03 INFO pred_bat: PV Data for pv_forecast_today total 26.84 kWh

Futurerate inactive
17:52:04 INFO pred_bat: Predbat mode is set to Control charge
17:52:04 INFO pred_bat: Found 21600 load_today datapoints going back 14 days
17:52:05 INFO pred_bat: Car charging hold True threshold 6.0
17:52:05 INFO pred_bat: Current data so far today: load 8.17 kWh import
11.8 kWh export 12.7 kWh pv 16.7 kWh
17:52:05 INFO pred_bat: Fetching futurerate data from None
17:52:05 INFO pred_bat: Standing charge is set to 60.01800000000001 p
17:52:06 INFO pred_bat: Rate min forward looking: now 10.58 at end of
forecast 10.58
17:52:06 INFO pred_bat: Import rates min 10.58 max 32.85 average 16.92
17:52:06 INFO pred_bat: Export rates min 15.0 max 15.0 average 15.0
17:52:06 INFO pred_bat: Rate thresholds (for charge/discharge) are import
32.35p (0.0) export 15.0p (0.0)
17:52:06 INFO pred_bat: High export rate found rates in range 15.0 to 15.0
17:52:06 INFO pred_bat: Low Import rate found rates in range 11.02 to 32.01
17:52:06 INFO pred_bat: Todays energy import 11.8 kWh export 12.7 kWh cost
-49.42 p import 141.07 p export -190.49 p
17:52:06 INFO pred_bat: PV Data for pv_forecast_today total 26.84 kWh

**Comment by springfall2008** [comment]
It must be failing to download the data, can you actually get to that web address on your HA machine?

**Comment by stjohncanning** [comment]
Well, I deleted PREDBAT and reinstalled program, reconfigured a new
apps.yaml, and now giving rates which it wasn't before!
No idea what was wrong, couldn't see anything wrong with the original apps
file, also settings are same as they were before so a bit of a mystery to
me but as everything seems to be working will check it out tomorrow when
the 'future rates' should appear in the graphs etc.
Thanks for your inputs, appreciated.

19:52:20 INFO pred_bat: Fetching futurerate data from
https://www.nordpoolgroup.com/api/marketdata/page/325?currency=GBP
19:52:21 INFO pred_bat: Future rates - peak rate is 0 - 0 minutes premium
import 0 export 0
19:52:21 INFO pred_bat: Loaded 192 datapoints of futurerate analysis
19:52:21 INFO pred_bat: Predicted future rates: ['04-19 19:00:00 => 19.99 /
8.22', '04-19 20:00:00 => 19.7 / 8.1', '04-19 21:00:00 => 19.81 / 8.15',
'04-19 22:00:00 => 16.63 / 6.84', '04-19 23:00:00 => 15.64 / 6.43', '04-20
00:00:00 => 14.13 / 5.81', '04-20 01:00:00 => 15.71 / 6.46', '04-20
02:00:00 => 16.32 / 6.71', '04-20 03:00:00 => 15.6 / 6.42', '04-20 04:00:00
=> 14.02 / 5.77', '04-20 05:00:00 => 14.09 / 5.79', '04-20 06:00:00 =>
16.34 / 6.72', '04-20 07:00:00 => 17.72 / 7.29', '04-20 08:00:00 => 17.52 /
7.2', '04-20 09:00:00 => 17.09 / 7.03', '04-20 10:00:00 => 16.19 / 6.66',
'04-20 11:00:00 => 15.03 / 6.18', '04-20 12:00:00 => 15.75 / 6.48', '04-20
13:00:00 => 13.63 / 5.61', '04-20 14:00:00 => 12.13 / 4.99', '04-20
15:00:00 => 11.82 / 4.86', '04-20 16:00:00 => 15.62 / 6.42', '04-20
17:00:00 => 17.56 / 7.22', '04-20 18:00:00 => 19.88 / 8.18', '04-20
19:00:00 => 20.78 / 8.54', '04-20 20:00:00 => 20.79 / 8.55', '04-20
21:00:00 => 18.43 / 7.58', '04-20 22:00:00 => 16.3 / 6.7']
19:52:21 INFO pred_bat: Standing charge is set to 60.01800000000001 p

**Comment by stjohncanning** [comment, marked as answer]
Confirmed working as designed today.  Can only conclude something not right with the original installation as reinstall has fixed it.  Closed
